player/public/coreplugins/org.ekstep.videorenderer-1.1/renderer/plugin.js

```diff
@@ -0,0 +1,324 @@
+    enableHeartBeatEvent:false,
+    _constants: {
+        mimeType: ["video/mp4", "video/x-youtube", "video/webm"],	
+        events: {	
+            launchEvent: "renderer:launch:video"	
+        }	
+    },	
+    initLauncher: function() {	
+        EkstepRendererAPI.addEventListener(this._constants.events.launchEvent, this.start, this);
+        EkstepRendererAPI.addEventListener("renderer:overlay:mute", this.onOverlayAudioMute, this);
+        EkstepRendererAPI.addEventListener("renderer:overlay:unmute", this.onOverlayAudioUnmute, this);
+    },
+    start: function() {
+        this._super();
+        var data = _.clone(content);
+        this.heartBeatData.stageId = content.mimeType === 'video/x-youtube' ? 'youtubestage' : 'videostage';
+        var globalConfigObj = EkstepRendererAPI.getGlobalConfig();
+        if (window.cordova || !isbrowserpreview) {
+            var prefix_url = globalConfigObj.basepath || '';
+            path = prefix_url + "/" + data.artifactUrl;
+        } else {
+            path = data.artifactUrl;
+        }
+        this.createVideo(path, data);
+        this.configOverlay();
+    },
+    createVideo: function(path, data) {
+        EkstepRendererAPI.dispatchEvent("renderer:splash:hide");
+
+        video = document.createElement('video-js');
+        video.style.width = '100%';
+        video.style.height = '100%';
+        video.style.position = 'absolute';
+        video.id = "videoElement";
+        video.className = 'vjs-default-skin';
+        document.body.appendChild(video);
+        EkstepRendererAPI.dispatchEvent("renderer:content:start");
+
+        //Todo need to remove after testing
+        //data.previewUrl = "https://sunbirdspikemedia-inct.streaming.media.azure.net/afcc5a99-d0c4-4ef5-9dfe-dc403a1269fb/learn-colors-with-numbers-in-kid.ism/manifest(format=m3u8-aapl-v3)";
+
+        if(data.mimeType === "video/x-youtube"){
+            this._loadYoutube(data.artifactUrl);
+        }else if(this.isStreaming && (data.mimeType != "video/x-youtube")){
+            data.mimeType = this.supportedStreamingMimeType;
+            this._loadVideo(data.previewUrl, data);
+        }else{
+            this._loadVideo(path, data);
+        }
+        
+        video.onvolumechange = function() {
+            (video.muted) ? EkstepRendererAPI.dispatchEvent('renderer:overlay:mute') : EkstepRendererAPI.dispatchEvent('renderer:overlay:unmute');
+        };
+        $("video").bind("contextmenu",function() {
+            return false;
+        });
+    },    
+    _loadVideo: function(path, data) {
+        var instance = this;
+        if (instance.isStreaming && !navigator.onLine) {
+            EkstepRendererAPI.logErrorEvent('No internet', {
+                'type': 'content',
+                'action': 'play',
+                'severity': 'error'
+            });
+            instance.throwError({message: instance.messages.noInternetConnection});
+            if (typeof cordova !== "undefined") exitApp();
```

Comment:
why are doing force `exitApp();` ?

Reply:
exitApp is not required i guess. Either it should re-connect to internet and start play video or let user to exit the player.